test/p3_tests.jl

```diff
@@ -448,31 +458,44 @@ function test_numerical_integrals(FT)
             # The `rtol` settings essentially forces max evaluations of the method.
             # Note 2: For F_rim=0, L=0.002, even higher order quadrature rules are needed.
             N′ = P3.size_distribution(state, logλ)
-            N_estim = P3.∫fdD(state, logλ) do D
-                N′(D)
-            end
-            @test N_ice ≈ N_estim rtol = 1e-5
+            bnds = P3.integral_bounds(state, logλ; p = 1e-6, moment_order = 0)
+            N_estim_cheb = P3.integrate(N′, bnds...)
+            @test N_ice ≈ N_estim_cheb rtol = 1e-5
+
+            # Compare with quadgk
+            N_estim_qgk = QGK.quadgk(N′, bnds...)[1]
+            @test N_estim_cheb ≈ N_estim_qgk rtol = 1e-5
+
 
             # Bulk velocity comparison
             vel_N = P3.ice_terminal_velocity_number_weighted(Chen2022, ρ_a, state, logλ; use_aspect_ratio, p)
             vel_m = P3.ice_terminal_velocity_mass_weighted(Chen2022, ρ_a, state, logλ; use_aspect_ratio, p)
 
             v_term = P3.ice_particle_terminal_velocity(Chen2022, ρ_a, state; use_aspect_ratio)
             g(D) = v_term(D) * N′(D)
-            vel_N_estim = P3.∫fdD(g, state, logλ; p) / N_ice
-            vel_m_estim = P3.∫fdD(state, logλ; p) do D
-                g(D) * P3.ice_mass(state, D) / L_ice
-            end
+            gm(D) = g(D) * P3.ice_mass(state, D)
+            vel_N_estim_cheb = P3.integrate(g, bnds...; quad = P3.ChebyshevGauss(10)) / N_ice
+            vel_m_estim_cheb = P3.integrate(gm, bnds...; quad = P3.ChebyshevGauss(10)) / L_ice
+            @test vel_N ≈ vel_N_estim_cheb rtol = 0.005
+            @test vel_m ≈ vel_m_estim_cheb rtol = 0.05
```

Comment:
Do you know why the tolerance increased so badly in those tests? And should we care?